src/main/webapp/scripts/view_card.js

```diff
@@ -13,42 +13,81 @@
+    radius: Number(vendor.saleCard.location.radius),
+    markerName: vendor.saleCard.businessName,
+    type: 'vendor',
+    drawCircleRadius: vendor.saleCard.hasDelivery
+  };
 
+  showVendorInfo(vendor);
+  drawMap(vendorInfo);
+}
+
+/**
+ * Shows vendor's information in the page.
+ * @param {object} vendor Retrieved from servlets.
+ */
+const showVendorInfo = (vendor) => {
   const salecardTemplate = document.getElementById('salecard-template');
   const salecardContainer = document.getElementById('salecard-container');
 
-  //Clean previously retrieved cards
+  // Clean previously retrieved cards.
   salecardContainer.textContent = '';
 
+  let salecard = vendor.saleCard;
   let salecardCloned = salecardTemplate.content.cloneNode(true);
 
+  salecardCloned.getElementById('business-picture').src
+      = `/serve-blob?blobKey=${vendor.saleCard.picture.blobKey.blobKey}`;
+  salecardCloned.getElementById('business-picture').alt = vendor.saleCard.picture.altText;
   salecardCloned.getElementById('salecard-id').textContent = salecard.id;
   salecardCloned.getElementById('business-name').textContent = salecard.businessName;
   salecardCloned.getElementById('business-description').textContent = salecard.description;
-  salecardCloned.getElementById('openedFrom').textContent = 
-    `${salecard.startTime.hour}:${salecard.startTime.minute}:${salecard.startTime.second}`;
-  salecardCloned.getElementById('openedTo').textContent = 
-    `${salecard.endTime.hour}:${salecard.endTime.minute}:${salecard.endTime.second}`;
-  salecardCloned.getElementById('deliveryDistance').textContent = salecard.location.radius;
+  salecardCloned.getElementById('openedFrom').textContent = parseTime(salecard.startTime);
+  salecardCloned.getElementById('openedTo').textContent = parseTime(salecard.endTime);
+  salecardCloned.getElementById('vendor-distance').textContent = `${salecard.distanceFromClient.toFixed(2)}m`;
   salecardCloned.getElementById('vendor-name').textContent = `${vendor.firstName} ${vendor.lastName}`;
   salecardCloned.getElementById('vendor-phone').textContent = vendor.phoneNumber;
-  salecardCloned.getElementById('vendor-distance').textContent = `${salecard.distanceFromClient}m`;
   salecardCloned.getElementById('business-lat').value = salecard.location.salePoint.latitude;
   salecardCloned.getElementById('business-lng').value = salecard.location.salePoint.longitude;
-  salecardContainer.appendChild(salecardCloned);
 
-  // After the card is inserted on the body we can init the map
-  initMap();
+  if (salecard.hasDelivery) {
+    salecardCloned.getElementById('deliveryDistance').textContent =
+        `${salecard.location.radius}m`;
+  } else {
+    const deliveryElement = salecardCloned.getElementById('deliveryElement');
+    deliveryElement.classList.remove('alert-primary');
+    deliveryElement.classList.add('alert-danger');
+    deliveryElement.textContent = 'This vendor does not offer delivery service';
```

Comment:
here as well, and the "{salecard.location.radius}m" above too. The distance one is a bit tricky. I think it's risky to leave both the content and the units in-line. If suddenly you wanted to add the ability to change from meters to km or to miles, you'd have to track down all these little m's in your files. 

Can JS make strings with placeholders in them, like on Java this might be "%s m" and then the "%s" is replaced by whatever string I provide during parsing. Somehow abstracting that into a method that can take a distance and maybe a DistanceType enum value, and return stylized content could be a good exercise, even if you don't expose to the user the ability to change that enum selection, it would still be there in the background for when you add it to the UI.

@akihiroota87 for opinions on this.

Reply:
Here's some documentation on string literals in JS: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

To my knowledge this is the only way to insert arguments into a string (aside from concatenation).

Reply:
Yes, I think implement unit conversion on our platform will be great, but for making that change we need to support that on our servlets too, I'll discuss this with the pod in our meeting and see what they think. I'll merge for now, but this will be in the list of features.